src/app/notebook/notebook.component.ts

```diff
@@ -0,0 +1,103 @@
+import { Component, Input, OnChanges } from '@angular/core';
+import { finalize } from 'rxjs/operators';
+import { Subscription } from 'rxjs';
+import { SourceFile, Workflow, WorkflowVersion, WorkflowsService } from 'app/shared/openapi';
+
+/**
+ * Convert the specified notebook to user-friendly preview that represents it.
+ *
+ * The Jupyter notebook format:
+ * https://nbformat.readthedocs.io/en/latest/
+ */
+
+@Component({
+  selector: 'app-notebook',
+  templateUrl: './notebook.component.html',
+})
+export class NotebookComponent implements OnChanges {
+  constructor(private workflowsService: WorkflowsService) {}
+  @Input() notebook: Workflow;
+  @Input() version: WorkflowVersion;
+  @Input() baseUrl: string;
+  loading = true;
+  error = false;
+  cells = [];
+  private currentSubscription: Subscription = null;
+
+  ngOnChanges() {
+    this.retrieveAndFormatNotebook();
+  }
+
+  retrieveAndFormatNotebook() {
+    this.loading = true;
+    this.error = false;
+    this.cells = [];
+    // The next line cancels any previous request that is still in progress,
+    // because if we're here, the @Inputs have changed, and we're about to
+    // launch a new request to retrieve the corresponding notebook file,
+    // which we will display when it arrives.  The previous response is now
+    // irrelevant, and could actually cause problems if it arrives out of
+    // order.
+    // Assuming that Angular calls a component's `ngOnChanges()` directly
+    // after setting its @Input fields, a pleasant side effect of said
+    // cancellation is that when we handle a response, we know that @Input
+    // fields have the same values as when the corresponding request was
+    // started, which simplifies the code.  For example, the methods
+    // called by the handler to format the notebook can safely use the
+    // value of the `notebook` field to determine the notebook language.
+    this.currentSubscription?.unsubscribe();
+    this.currentSubscription = this.workflowsService
```

Comment:
We're not always consistent, but we generally use the AlertService for Ajax requests (and the error).

Reply:
Yeah, this code is similar to that behind the Files tab, which doesn't use AlertService either.  Is the alert functionality redundant with the error message that appears where the preview should be?